lib/clearance/session.rb

```diff
@@ -115,11 +115,28 @@ def cookie_value
         value: remember_token
       }
 
-      if Clearance.configuration.cookie_domain.present?
-        value[:domain] = Clearance.configuration.cookie_domain
+      allowed_domains = Array(Clearance.configuration.cookie_domain).dup
+      allowed_domains.map!(&:downcase)
+
+      if allowed_domains.any?
+        base_host = extract_base_host(get_request_uri)
+        value[:domain] = allowed_domains.delete(base_host)
+        value[:domain] ||= allowed_domains.first
       end
 
       value
     end
+
+    private
+
+    def extract_base_host(uri)
+      hostname = URI.parse(uri).host
+      hostname.split(".").last(2).join(".").downcase
```

Comment:
What about the need to set cookies based on subdomain? That is: `thoughtbot.myapp.com`?